lib/IDE/TypeReconstruction.cpp

```diff
@@ -1906,6 +1906,30 @@ static void VisitNodeInOut(
   }
 }
 
+static void VisitNodeExistentialMetatype(ASTContext *ast,
+                                         Demangle::NodePointer cur_node,
+                                         VisitNodeResult &result) {
+  VisitNodeResult type_result;
+  Optional<MetatypeRepresentation> metatype_repr;
+
+  for (auto &child : *cur_node) {
+    switch (child->getKind()) {
+    case Demangle::Node::Kind::Type:
+      VisitNode(ast, child, type_result);
+      break;
+    default:
+      break;
+    }
+  }
+
+  if (type_result.HasSingleType())
+    result._types.push_back(
+        ExistentialMetatypeType::get(type_result._types[0], metatype_repr));
```

Comment:
Should you check that type_result._types is nonempty?

Reply:
I think the non-emptiness it's already guaranteed as part of `HasSingleType`, which is defined as:
```
bool HasSingleType() { return _types.size() == 1 && _types.front(); }
```